store/sqlstore/group_supplier.go

```diff
@@ -919,25 +934,35 @@ func (s *SqlSupplier) ChannelMembersToRemove(ctx context.Context, hints ...store
 	return result
 }
 
-func (s *SqlSupplier) groupsByTeamBaseQuery(t selectType, teamID string, opts model.GroupSearchOpts) squirrel.SelectBuilder {
+func (s *SqlSupplier) groupsBySyncableBaseQuery(st model.GroupSyncableType, t selectType, syncableID string, opts model.GroupSearchOpts) squirrel.SelectBuilder {
 	selectStrs := map[selectType]string{
 		selectGroups:      "ug.*",
 		selectCountGroups: "COUNT(*)",
 	}
 
+	var table string
+	var idCol string
+	if st == model.GroupSyncableTypeTeam {
+		table = "GroupTeams"
+		idCol = "TeamId"
+	} else {
+		table = "GroupChannels"
+		idCol = "ChannelId"
+	}
+
 	query := s.getQueryBuilder().
 		Select(selectStrs[t]).
-		From("GroupTeams gt").
-		LeftJoin("UserGroups ug ON gt.GroupId = ug.Id").
-		Where("ug.DeleteAt = 0 AND gt.TeamId = ? AND gt.DeleteAt = 0", teamID)
+		From(fmt.Sprintf("%s gs", table)).
+		LeftJoin("UserGroups ug ON gs.GroupId = ug.Id").
+		Where(fmt.Sprintf("ug.DeleteAt = 0 AND gs.%s = ? AND gs.DeleteAt = 0", idCol), syncableID)
```

Comment:
I don't like this "query building" approach. I would prefer something more simple like building the entire query two times one for each syncable type. (probably in the query some lines later you can reuse a chunk of the query. Anyway, up to you, feel free to merge it as is, if you think is the best option.

Reply:
@jespino I had the same thought as you when it was being written. Perhaps it's tipping the scale towards "too clever" instead of being repetitive. The impetus for doing it this way was that it's such a long query to repeat twice. Is the risk from the repetition worth the risk from the added complexity of interpolating the string? I don't think there's a definitive answer.